pkg/kubernetes/jobs.go

```diff
@@ -74,20 +76,53 @@ func (k *KubeJobManager) StartJobCleaner(ctx context.Context, wg *sync.WaitGroup
 }
 
 // LaunchJob launches a Kubernetes job for a given Buildkite job ID
-func (k *KubeJobManager) LaunchJob(uuid string) error {
+func (k *KubeJobManager) LaunchJob(job *buildkite.Job) error {
 	var err error
+	var t batchv1.Job
+	uuid := *job.ID
 
 	jobLabels := make(map[string]string)
 	jobLabels["kubekite-managed"] = "true"
 	jobLabels["kubekite-org"] = k.org
 	jobLabels["kubekite-pipeline"] = k.pipeline
 
-	t := k.jobTemplate
+	err = json.Unmarshal(k.jobTemplate, &t)
+	if err != nil {
+		log.Fatal(err)
+	}
 
 	// Set our labels on both the job and the pod that it generates
 	t.SetLabels(jobLabels)
 	t.Spec.Template.SetLabels(jobLabels)
 
+	container := t.Spec.Template.Spec.Containers[0]
+
+	tags_var := corev1.EnvVar{"BUILDKITE_AGENT_TAGS", strings.Join(job.AgentQueryRules, ","), nil}
+	container.Env = append(container.Env, tags_var)
+	for _, v := range job.AgentQueryRules {
+		split_rule := strings.Split(v, "=")
+		if len(split_rule) != 2 {
```

Comment:
Potential bug: if the rule is something like `image=mybase64encodednonsense==` you'll end up with more than two elements, even though you would want to actually join everything on the tail of the list.